benchmark/single-source/SetTests.swift

```diff
@@ -12,69 +12,115 @@
 
 import TestsUtils
 
+let size = 400
+let overlap = 100
+
+let setAB = Set(0 ..< size)
+let setCD = Set(size ..< 2 * size)
+let setBC = Set(size - overlap ..< 2 * size - overlap)
+let setB = Set(size - overlap ..< size)
+
+let setOAB = Set(setAB.map(Box.init))
+let setOCD = Set(setCD.map(Box.init))
+let setOBC = Set(setBC.map(Box.init))
+let setOB = Set(setB.map(Box.init))
+
+let countAC = 2 * (size - overlap)
+let countABC = 2 * size - overlap
+let countABCD = 2 * size
+let countB = overlap
+
 public let SetTests = [
-  BenchmarkInfo(name: "SetExclusiveOr2", runFunction: run_SetExclusiveOr2, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetExclusiveOr2_OfObjects", runFunction: run_SetExclusiveOr2_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIntersect2", runFunction: run_SetIntersect2, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIntersect2_OfObjects", runFunction: run_SetIntersect2_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIsSubsetOf2", runFunction: run_SetIsSubsetOf2, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIsSubsetOf2_OfObjects", runFunction: run_SetIsSubsetOf2_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetUnion2", runFunction: run_SetUnion2, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetUnion2_OfObjects", runFunction: run_SetUnion2_OfObjects, tags: [.validation, .api, .Set]),
+  BenchmarkInfo(
+    name: "SetExclusiveOr2",
+    runFunction: { n in run_SetExclusiveOr(setAB, setBC, countAC, 10 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setBC]) }),
+  BenchmarkInfo(
+    name: "SetExclusiveOr2_OfObjects",
+    runFunction: { n in run_SetExclusiveOr_OfObjects(setOAB, setOBC, countAC, 10 * n) },
```

Comment:
I'm skeptical that the same multiplier for objects as for `Int`s woudn't overshoot the healthy runtime. Can we aim for 1ms (1000 μs)?

Reply:
The difference between `Int` and `Box<Int>` is relatively small; it's merely an extra indirection. It is important to remain able to keep track of the difference, though.

Capitalized names for function parameters is a silly tradition; I'm happy to be the barbarian who breaks it. ;-)

Reply:
(Oh, beyond the indirection, Box<Int> also hashes a bit slower than Int since it's not forwarding the one-shot hashing implementation.)